library/kernel_settings.py

```diff
@@ -0,0 +1,990 @@
+      to names of units or plugins in tuned.  For example, the option `sysctl`
+      corresponds to the `sysctl` unit or plugin in tuned.  Setting parameters
+      works mostly like it does with tuned, except that this module uses Ansible
+      YAML format instead of the tuned INI-style profile file format.  This module
+      creates a special tuned profile `kernel_settings` which will be applied by
+      tuned before any other profiles, allowing the user to configure tuned to
+      override settings made by this module.  You should be aware of this if you
+      plan to use tuned in addition to using this module.
+      NOTE: the options list may be incomplete - the actual options are generated
+      dynamically from tuned, for the current options supported by the version
+      of tuned, which are the tuned supported plugins.  Only the most common
+      options are listed.  See the tuned documentation for the full list and
+      more information.
+
+options:
+    sysctl:
+        description:
+            - key/value pairs of sysctl settings to apply - this works mostly
+              like the `sysctl` module except that `/etc/sysctl.conf` and files
+              under `/etc/sysctl.d` are not used.
+        required: false
+    sysfs:
+        description:
+            - key/value pairs of sysfs settings to apply
+        required: false
+    bootloader:
+        description:
+            - the `cmdline` option can be used to set, add, or delete
+              kernel command line options.  See EXAMPLES for some examples
+              of how to use this option.
+              Note that this uses the tuned implementation, which adds these
+              options to whatever the default bootloader command line arguments
+              tuned is historically used to add/delete performance related
+              kernel command line arguments e.g. `spectre_v2=off`.  If you need
+              more general purpose bootloader configuration, you should use
+              a bootloader module/role.
+    operations:
+        description:
+            - This a dict.  Each key is the name of a section to operate on -
+              `sysctl`, `sysfs`, etc. same as the module setting options.  The
+              value is a list of operations to apply.  Each individual operation
+              is a dict.  Mostly this is used to delete values from sections i.e.
+              to restore their default values.  You can delete a value, then add
+              it back, to avoid having the role issue a `Conflict` warning that a
+              value conflicts with an existing value.  See `EXAMPLES` below.
+    purge:
+        description:
+            - Remove the current kernel_settings, whatever they are, and force
+              the given settings to be the current and only settings
+
+author:
+    - Rich Megginson (rmeggins@redhat.com)
+"""
+
+EXAMPLES = """
+# add the sysctl vm.max_mmap_regions, and disable spectre/meltdown security
+kernel_settings:
```

Comment:
`kernel_settings_parameters`?

Reply:
No, this is for the module, not the role.  The name of the module is `kernel_settings`.